Microsoft.Toolkit.Uwp.UI/Triggers/IsEqualStateTrigger.cs

```diff
@@ -92,14 +94,14 @@ private static bool ConvertTypeEquals(object value1, object value2)
 
         private static object ConvertToEnum(Type enumType, object value)
         {
-            try
+            // value cannot be the same type of enum now
+            return value switch
             {
-                return Enum.IsDefined(enumType, value) ? Enum.ToObject(enumType, value) : null;
-            }
-            catch
-            {
-                return null;
-            }
+                string str => Enum.TryParse(enumType, str, out var e) ? e : null,
+                int or uint or byte or sbyte or long or ulong or short or ushort
+                    => Enum.ToObject(enumType, value),
```

Comment:
Minor note: now that we no longer have `Enum.IsDefined`, this PR introduces a small behavior change. As in, if `value` happens to be a boxed number that is outside of the defined range for the current enum type, this will now still succeed and just return an undefined (boxed) enum value, whereas before this PR it'd return `null` instead.

I mean, this should likely not happen anyway, but just worth pointing it out.
I'm fine with just leaving up to consumers to ensure input raw numeric values are in fact in range.

Reply:
I think that's fine. As a StateTrigger this should be somethings Devs are working with in a VSM, so it'd be a bit odd to try and use something that wasn't defined?

Reply:
Yup agreed, just figured I'd point that out to clarify since it's effectively a (small) functional change though 🙂